src/services/GooglePlacesService.ts

```diff
@@ -131,6 +163,17 @@ class GooglePlacesService {
         return Promise.reject(err);
       });
   }
+
+  // helper function used to refresh place ID and call getRestaurantDetails again for NOT_FOUND error
+  refreshPlaceIDAndGetDetails(placeID: string): Promise<RestaurantDetails> {
+    return this.refreshPlaceID(placeID)
+      .then((newPlaceID: string) => {
+        return this.getRestaurantDetails(newPlaceID);
+      })
```

Comment:
Hmm, I'm kind of scared of this recursive call resulting in a stack overflow error

Reply:
I think one way we could approach handling this case is by changing the signature of `refreshPlaceID` to include a `handleNotFoundError` optional parameter which would be a callback. And then we can extract

```
 if (restaurantData.status && restaurantData.status === 'NOT_FOUND') {
    return this.refreshPlaceIDAndGetDetails(placeID);
 }
```

into a private helper function and replace it with a call to that helper function (and also we would use this function as a default for this parameter). Then here on line 171, we could just pass in a callback that doesn't include a call to `refreshPlaceIDAndGetDetails`, terminating the recursion?